### ISSUE TITLE
DataFilterExtension (category) and PathStyleExtension not working together

### ISSUE THREAD

**Issue by AdityaC-Lepton** [visgl/deck.gl#9253]
### Description

When added PathStyleExtension with DataFilterExtension filtering with category the line is not visible however it woks when I use range filter

**Before adding DataFilterExtension** 
![image](https://github.com/user-attachments/assets/96332619-e113-4c81-bb40-a4c9e07d210d)

![image](https://github.com/user-attachments/assets/058f55f5-0944-4628-ad77-255a0a543ac9)

**After adding DataFilterExtension** 

![image](https://github.com/user-attachments/assets/f9bb8f11-0d11-4f0a-9a94-596ff1becd36)

![image](https://github.com/user-attachments/assets/d9ed8724-3a07-4cc1-8f2c-a603b8822525)

### Flavors

- [ ] Script tag
- [ ] React
- [ ] Python/Jupyter notebook
- [ ] MapboxOverlay
- [X] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

Line should be visible using both extension

### Steps to Reproduce

   return new GeoJsonLayer({
       id: 'cable',
       useDevicePixels: app.useDevicePixelsInVectorLayer,
           data: app.filterDataWithProvinceGeom(app.cableFilteredGeoJson, "FeatureCollection","CBL"), //app.cableFilteredGeoJson,
       lineWidthScale: 1,
       lineWidthUnits: 'pixels',
       lineWidthMinPixels: 1,
       lineWidthMaxPixels: 10,
       getLineColor: f => app.GetLineColor(f),
       getLineWidth: feature => app.GetLineWidth(feature),
       opacity: styleObj[0].LayerStyle[0].opacity,
       autoHighlight: true,
       pickable: true,
       visible: app.isVectorLayerActive("CBL"),
       getFilterCategory: f => f.properties.network_status,
       filterCategories: app.GetfilterCategories("CBL"),
       extensions: [new PathStyleExtension({ highPrecisionDash: false, dash: true, dashJustified: true }), new DataFilterExtension({ categorySize: 1 })],
       getDashArray: f => (app.GetDashArray(f.properties.network_status)),
       collisionEnabled: app.IsCollisionEnabled,

### Environment

- Framework version: deck.gl@9.0.35
- Browser: Chrome 130.0.6723.117
- OS: Windows 11

**Comment by fnberta** [comment]
I'm hitting this as well. @AdityaC-Lepton did you find a workaround?

**Comment by AdityaC-Lepton** [comment]
No, I’m using the range filter. I’ve updated my code to work with it accordingly.
